ArcGISRuntimeSDKQt_QMLSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.qml

```diff
@@ -0,0 +1,279 @@
+                    style: Enums.SimpleLineSymbolStyleSolid
+                    color: "red"
+                    width: 5
+                }
+            }
+        }
+
+        OfflineMapTask {
+            id: offlineMapTask
+            onlineMap: arcgisOnlineMap
+
+            onErrorChanged: console.log(error.message, error.additionalMessage);
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                // fetch preplanned map areas from the portal item
+                offlineMapTask.preplannedMapAreas();
+                path = outputMapPackage;
+                fileFolder.url = path;
+                const worked = fileFolder.makeFolder();
+                if (!worked)
+                    console.log("Make Folder Failed");
+            }
+
+            onPreplannedMapAreasStatusChanged: {
+                if (preplannedMapAreasStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                preplannedCombo.model = offlineMapTask.preplannedMapAreaList;
+                busy.visible = false;
+
+                for (let i = 0; i < offlineMapTask.preplannedMapAreaList.count; i++) {
+                    let mapArea = offlineMapTask.preplannedMapAreaList.get(i);
+                    mapArea.loadStatusChanged.connect(function () {
+                        if (offlineMapTask.preplannedMapAreaList.get(i).loadStatus !== Enums.LoadStatusLoaded)
+                            return;
+
+                        const graphic = ArcGISRuntimeEnvironment.createObject("Graphic", { geometry: offlineMapTask.preplannedMapAreaList.get(i).areaOfInterest });
+
+                        graphicsOverlay.graphics.append(graphic);
+                    });
+                    offlineMapTask.preplannedMapAreaList.get(i).load();
+                }
+            }
+
+            onCreateDefaultDownloadPreplannedOfflineMapParametersStatusChanged: {
+                if (createDefaultDownloadPreplannedOfflineMapParametersStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                /* Set the update mode to not receive updates.
+                 * Other options:
+                 * SyncWithFeatureServices - changes will be synced directly with the
+                 * underlying feature services.
+                 * DownloadScheduledUpdates - schedulded, read-only updates will be
+                 * downloaded and applied to the local geodatabase. */
+                createDefaultDownloadPreplannedOfflineMapParametersResult.updateMode = Enums.PreplannedUpdateModeNoUpdates;
+                let result = createDefaultDownloadPreplannedOfflineMapParametersResult;
+
+                path = outputMapPackage +"/" + result.preplannedMapArea.portalItem.title;
+                fileFolder.url = path;
+
+                if (fileFolder.exists) {
+                    var mmpk = ArcGISRuntimeEnvironment.createObject("MobileMapPackage", { path: path });
```

Comment:
might be good to cache this as a global variable. Seems like you'd want the MMPK to persist throughout the lifetime of the sample?